minichain/p2p.py

```
@@ -1,390 +1,176 @@
-
     async def start(self, port: int = 9000, host: str = "127.0.0.1"):
-        """Start listening for incoming peer connections on the given port."""
-        self._port = port
-        self._server = await asyncio.start_server(self._handle_incoming, host, port)
-        logger.info("Network: Listening on %s:%d", host, port)
+        self.port = port
+        self.host_addr = host
+        self.loop = asyncio.get_running_loop()
+        
+        threading.Thread(target=trio.run, args=(self._trio_main,), daemon=True).start()
+        asyncio.create_task(self._asyncio_reader())
```

Ruff (0.15.17)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)